packages/api-kit/src/SafeApiKit.ts

```diff
@@ -715,6 +719,110 @@ class SafeApiKit {
       }
     })
   }
+
+  /**
+   * Get the SafeOperations that were sent from a particular address.
+   * @param safeAddress - The Safe address to retrieve SafeOperations for
+   * @throws "Safe address must not be empty"
+   * @throws "Invalid Ethereum address {safeAddress}"
+   * @returns The SafeOperations sent from the given Safe's address
+   */
+  async getSafeOperationsByAddress(safeAddress: string): Promise<GetSafeOperationListResponse> {
+    if (!safeAddress) {
+      throw new Error('Safe address must not be empty')
+    }
+    const { address } = this.#getEip3770Address(safeAddress)
+
+    return sendRequest({
+      url: `${this.#txServiceBaseUrl}/v1/safes/${address}/safe-operations/`,
+      method: HttpMethod.Get
+    })
+  }
+
+  /**
+   * Get a SafeOperation by its hash.
+   * @param safeOperationHash The SafeOperation hash
+   * @throws "SafeOperation hash must not be empty"
+   * @throws "Not found."
+   * @returns The SafeOperation
+   */
+  async getSafeOperation(safeOperationHash: string): Promise<SafeOperationResponse> {
+    if (!safeOperationHash) {
+      throw new Error('SafeOperation hash must not be empty')
+    }
+
+    return sendRequest({
+      url: `${this.#txServiceBaseUrl}/v1/safe-operations/${safeOperationHash}/`,
+      method: HttpMethod.Get
+    })
+  }
+
+  /**
+   * Create a new 4337 SafeOperation for a Safe.
+   * @param addSafeOperationProps - The configuration of the SafeOperation
+   * @throws "Safe address must not be empty"
+   * @throws "Invalid Safe address {safeAddress}"
+   * @throws "Module address must not be empty"
+   * @throws "Invalid module address {moduleAddress}"
+   * @throws "SafeOperation is not signed by the given signer {signerAddress}"
+   */
+  async addSafeOperation({
+    moduleAddress: moduleAddressProp,
+    safeAddress: safeAddressProp,
+    safeOperation,
```

Comment:
So... I dont understand the use case for this particular function but here is some food for thought. Right now, there is no internal coupling between this and the Safe4337 module which is better then if we were importing the type from there. There is, however, an implicit one. For someone to use this endpoint, they would have to either: use `Safe4337Pack#createTransaction` or implementing an interface themselves that accomplishes the same thing (which is a bit awkward). Are there situation where all these things could be supplied by an external library or people should ALWAYS use both modules together? Should we really use such complex object just for an api proxy and should the api-kit itself sign in this case? Perhaps the interface here is just really an `UserOperation` (which is clearly defined by the ERC) and the fields associated with packing signature (`validUntil`, `validAfter` and the authorizer's `signature`). That data field also almost look like something to represent the internal state of the object.